path_login.go

```diff
@@ -273,8 +274,27 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 			principalIDs[to.String(vmss.Identity.PrincipalID)] = struct{}{}
 		}
 		// if not, look for user-assigned identities
-		for _, userIdentity := range vmss.Identity.UserAssignedIdentities {
-			principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+		for userIdentityID := range vmss.Identity.UserAssignedIdentities {
```

Comment:
I'm wondering if we should check if the user-assigned identity has a non-nil principal ID first and fall back to looking it up using the MSI client. This way our current users relying on VMSS uniform won't be broken by needing to update the secrets engine credentials with an additional role assignment. The new MSI API call requires `Microsoft.ManagedIdentity/userAssignedIdentities/read`. We don't want to break our current VMSS uniform users by introducing the code that works for flex.

For example, we could do something like:
```go
// if not, look for user-assigned identities
for userIdentityID, userIdentity := range vmss.Identity.UserAssignedIdentities {
	// Principal ID is not nil for VMSS uniform orchestration mode
	if userIdentity.PrincipalID != nil {
		principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
		continue
	}

	// Principal ID is nil for VMSS uniform orchestration mode, so we 
	// must look up the user-assigned identity using the MSI client
	elements := strings.Split(userIdentityID, "/")
	if len(elements) < 9 {}
	...
}
```

Reply:
Agree, that's good point! I just changed the code based on your suggestion.